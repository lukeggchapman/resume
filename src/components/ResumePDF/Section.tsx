import React from 'react'
import { View, Text, StyleSheet } from '@react-pdf/renderer'

const styles = StyleSheet.create({
  container: {},
  title: {
    fontFamily: 'Oswald',
    fontSize: 16,
    fontWeight: 900,
    marginBottom: 5,
  },
})

interface SectionProps {
  title: string
  children: React.ReactNode
}

const Section = ({ title, children }: SectionProps) => (
  <View style={styles.container}>
    <Text style={styles.title}>{title}</Text>
    {children}
  </View>
)

export default Section
