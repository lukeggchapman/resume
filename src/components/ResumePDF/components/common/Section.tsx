import React from 'react'
import { View, Text } from '@react-pdf/renderer'

import typography from '../typography'

export interface SectionProps {
  title: string
  children: React.ReactNode
}

const Section = ({ title, children }: SectionProps) => (
  <View>
    <Text style={typography.h2}>{title}</Text>
    {children}
  </View>
)

export default Section
