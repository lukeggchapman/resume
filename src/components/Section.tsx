import React from 'react'
import { View, Text } from '@react-pdf/renderer'

import typography from './typography'

export interface SectionProps {
  title: string
  children: React.ReactNode
}

function Section({ title, children }: SectionProps) {
  return (
    <View>
      <Text style={typography.h2}>{title}</Text>
      {children}
    </View>
  )
}

export default Section
