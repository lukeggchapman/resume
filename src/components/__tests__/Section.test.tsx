import React from 'react'
import { shallow } from 'enzyme'

import Section, { SectionProps } from '../Section'

const baseProps: SectionProps = {
  title: 'titleTest',
  children: 'ReactNodeTest',
}

describe('Section', () => {
  const render = (props?: Partial<SectionProps>) =>
    shallow(<Section {...{ ...baseProps, ...props }} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
