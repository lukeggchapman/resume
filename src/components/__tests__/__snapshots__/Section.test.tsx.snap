// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "borderBottom": "1 solid #333",
        "fontFamily": "Oswald",
        "fontSize": 16,
        "fontWeight": 900,
        "marginBottom": 5,
      }
    }
  >
    titleTest
  </Text>
  ReactNodeTest
</View>
`;
